CompareSVDVectors.py

```diff
@@ -0,0 +1,154 @@
+	return distances
+
+# gives the standard deviation of dictances
+def StandardDeviation (distances):
+	return numpy.std(distances)
+
+def DistancesFrequencies (distances) :
+	return itemfreq(distances.values()) # return [[val,freq],[val2,freq2]] in sorted order from the smallest value
+
+
+def splitKeysCorpus(dictKeys) :
+	ucblkeys = dict()
+	istexkeys = dict()
+	countUcbl = 0
+	countIstex = 0 
+	for k in dictKeys :
+		if k.find("UCBL") == 0 : # found UCBL at 0 position
+			ucblkeys[k] = (dictKeys[k], countUcbl )
+			countUcbl += 1
+		else : 
+			istexkeys[k] = (dictKeys[k], countIstex) # (doc2vec id, id for the splitted corpus)
+			countIstex += 1
+	return (ucblkeys, istexkeys)
+
+def loadModel(ModelFile) :
+	return Doc2Vec.load(ModelFile)
+
+def extractMatrix(model, list_keys, which_index=1) : # be carreful to extract the matrix in the index order (0 for doc2vec, complete model, 1 for the selected one (ucbl or istex))
+	matrix = []
+	for i in sorted([ x[which_index] for x in list_keys.values() ]) :
+		matrix.append(model.docvecs["DOC_%s"%i])
+	return numpy.array(matrix)
+
+def saveDistances(distances, fileName, keys, cluster = "0") : # distances is a dictionnary with documents keys (SourceDOI) and distance to the barycenter of UCBL douments (or any other distance we chosed)
+	#distInOrder = dict()
+	f=open(fileName, "w")
+	for k in sorted(distances, key=distances.get) : # sort keys according to values	
+		#distInOrder[k] = distances[k]
+		if str(k) in keys :
+			f.write(str(k) + ", " + str(distances[k]) + " : " + keys[str(k)] + " : " + "Cluster" + str(cluster) + "\n")
+		else :
+			f.write(str(k) + ", " + str(distances[k]) + "\n")
+
+	f.close()
+
+def computeKmeans(matrix, nb_clusters = 4) :
+	k_means = cl.KMeans(n_clusters=nb_clusters)
+	k_means.fit(mucbl)
+	
+	return k_means
+
+#make the list of the index of documents of cluster number cluster_index, given by the user
+def ListSameCluster(cl_labels, cluster_index = 0) :
+	l = []
+	for i in range(len(cl_labels)) :
+		if cl_labels[i] == cluster_index :
+			l.append(i)
+	return l
+
+if __name__  == "__main__" :
+	parser = argparse.ArgumentParser()
+	parser.add_argument("--model", default='output_svd.pickle', type=str) # contains .json files
+	parser.add_argument("--model_keys", default='output_paragraph_inverse_index.pickle', type=str) # is a .json file
+	parser.add_argument("--output_dir", default="UCBLIstexResults/", type=str) # contains wikipedia text files
+
+	args = parser.parse_args()
+	model = args.model
+	model_keys = args.model_keys
+	outputdir = args.output_dir
+
+
+	f = open(model, 'r')
+	svdMatrix = pickle.load(f)
+	f.close()
+
+	keys = open(model_keys, 'r')
+	index_keys = pickle.load(keys)
+	keys.close()
+
+	if not os.path.exists(outputdir):
+		os.makedirs(outputdir)
+
+	output = outputdir+"testSvdUcblIstexDistancesCluster"
+	f = open(outputdir+"StandardDeviation.txt", "w")
+	f.close()
+	f = open(outputdir+"Clusters4MentalRotation.txt", "w")
+	f.close()
```

Comment:
I don't understand why we need to open and then close these files (line 135 and line 137) directly without using them. I can see in line 146 that you use the files as needed. So, I think it is better to remove these (line 134 to line 138)